### PR TITLE
Turese/matching refactoring

### PR DESCRIPTION
**Pull request by Turese**
Moved non-rendering logic from the controller (matchings_controller.rb) to the models (matching.rb and project.rb).

### CODE REVIEW

**Review by kev-chien — APPROVED** [review]
Looks good! Thanks for moving controller logic into model.

File: app/models/project.rb [hunk 1]
```diff
@@ -7,3 +7,42 @@ class Project < ActiveRecord::Base
...
+      if ranking.rank != 0
+        row = {"person_name": person, "timeslot": timeslot_formatted, "rank": ranking.rank}
+        preferences.push(row)
+      end
+    end
+
+    preferences
+end
+
+public def email_participants (subject, body)
```
Comment (kev-chien) [inline app/models/project.rb]: should call this `email_matching_to_participants` given there's another email method that does emailing of links to rank

**Review by kev-chien — CHANGES_REQUESTED** [review]
Looks good! One minor function naming concern

**Review by lawrencechen14 — APPROVED** [review]
Thanks for refactoring this! Seems like we should be good now.

File: app/models/project.rb (hunk 1 above)
Comment (kev-chien) [reply app/models/project.rb]: done

**Review by kev-chien — APPROVED** [review]

**Pull request merged**